doc/dev/TESTING.md

```diff
@@ -1,77 +1,117 @@
+ - `DATA` should be a file in `testing/data` directory, though adding a new file is possible
  - `ARGS` should be the F3D options to pass to the f3d executable, if any
  - `DEFAULT_LIGHTS` is expected when performing baselines comparison
 
-Once the new test has been added, configure and build F3D, then run the test once:
+Once the new test has been added, configure and build F3D, then run the test (-VV for verbose output):
 
 ```
-ctest -R TestName
+ctest -R TestName -VV
 ```
 
-The test will fail but an image output will be generated in you build in `Testing/Temporary/TestName.png`.
+The test will fail but an image output will be generated in the build directory, namely `build/Testing/Temporary/TestName.png`.
 Visually check that the generated file looks as expected, then add it to the F3D sources in `testing/baselines`.
 Rerun the test, it should now pass.
 
+### Recovering baselines from CI
+Occasionally you may need to recover a baseline from the CI. If this is required, create a PR and let the test run (fail) on CI. Check the actions run summary on Github and download the appropriate `baseline` archive. Extract it and navigate to the `build/Testing/Temporary/TestName.png`. Extract this file and add it to the `testing/baselines` directory.
+
+### Creating Interaction tests
+Sometimes you may contribute changes that affect how the end user interacts with F3D, example, toggling orthographic projection on/off or zooming in/out of the rendered image. These human interactions are simulated by interaction tests. F3D has the functionality to record human interactions such as mouse wheel scrolls, mouse movements as well as keypresses to a file. This is enabled by running 
+
+```
+f3d --interaction-test-record ./TestName.log
+```
+
+where
+- `TestName` should be the name of the test case.
+
+While recording the interaction, care should be taken to perform the minimum number of events to simulate the interaction. For example, if you want to simulate the number `5` being pressed, there should be no mouse events or other keypress events during the recording of the interaction. The interaction file is saved when `f3d` is exited.
+
+You can verify that your interaction test file is correct by running
+
+```
+f3d --interaction-test-play ./TestName.log
+```
+
+After you have verififed that the interaction file works as expected, copy it to `./testing/recordings`. The name of the interaction file should be the same as the test name. The interaction test case can be created by adding 
+
+```
+f3d_test(NAME TestName DATA datafile.ext INTERACTION DEFAULT_LIGHTS)
+```
+
+to `application/testing/CMakeLists.txt`
+
+ where
+ - `NAME` should be the name of the test, which must be unique
+ - `DATA` should be a file in `testing/data` directory, though adding a new file is possible
+ - `INTERACTION` signifies that this is an interaction test
+ - `DEFAULT_LIGHTS` is expected when performing baselines comparison
```

Comment:
maybe explain only interaction and refer to above for the rest ?

Reply:
Noted